Multihop workers doc (#2772)
hashicorp/boundary: website/content/docs/configuration/worker/pki-worker.mdx

workers. These fields are only valid for [KMS Workers][]. `name` and
`description` can only be set for PKI workers through the API.

# Multi-Hop Workers

PKI workers can be configured via the `initial_upstreams field` to either connect directly to the control plane or,
when set to the `public_addr` of a worker, to other workers.

> Multi-hop is when two or more workers are connected, as there are multiple “hops” from a worker to the controller.

Reviewer: ```suggestion
A multi-hop configuration is when two or more workers are connected, and there are multiple “hops” from a worker to the controller.
```

Author: thanks!

> Multi-hop introduces the concept of “upstream” and “downstream” workers. Viewing controllers as the “top” of a

Reviewer: ```suggestion
The multi-hop configuration introduces the concepts of “upstream” and “downstream” workers. If you view controllers as the “top” of a
```

Author: thanks!

> multi-hop chain, downstream workers are those below a worker in the chain, while upstreams are those above a worker

Reviewer: ```suggestion
multi-hop chain, downstream workers reside below a worker in the chain, while upstream workers reside above a worker
```

Author: thanks!

> in the chain. For example, using the diagram below, Worker2’s upstream is Worker1, and its downstream is Worker3.

Reviewer: ```suggestion
in the chain. For example, in the diagram below, Worker2’s upstream is Worker1, and its downstream is Worker3.
```

Author: thanks!

![multi-hop workers](/img/multihop-workers.png)

> Multi-hop workers can be deployed in scenarios where inbound network traffic is not allowed. A worker in a private

Reviewer: ```suggestion
You can deploy multi-hop workers in scenarios where inbound network traffic is not allowed. A worker in a private
```

Author: thanks!

> network will send outbound communication to its upstream worker, and can create reverse proxies for session establishment.

Reviewer: ```suggestion
network can send outbound communication to its upstream worker, and create a reverse proxy to establish a session.
```

Author: thanks!

> [Target worker filters][] can be used with multi-hop workers to allow for fine grained control on which workers handle

Reviewer: ```suggestion
You can configure [target worker filters][] with multi-hop workers to allow for fine-grained control on which workers handle
```

Author: thanks!

> ingress and egress for session traffic to a [target][]. Ingress worker filters<sup>HCP Only</sup> determine which
> workers users will connect with to initiate a session, and egress worker filters determine which workers will be

Reviewer: ```suggestion
ingress and egress for session traffic to a [target][]. Ingress worker filters<sup>HCP Only</sup> determine which
workers you connect with to initiate a session, and egress worker filters determine which workers are
```

Author: thanks!

workers. These fields are only valid for [KMS Workers][]. `name` and
`description` can only be set for PKI workers through the API.

> # Multi-Hop Workers

Reviewer 2: ```suggestion
# Multi-Hop Workers<sup>HCP Only</sup>
```

Reviewer 2: Adding the "HCP Only" tag as the multi hop session functionality is only for HCP users

You can use the `initial_upstreams` field to configure PKI workers to either connect directly to the control plane
or to other workers, when set to the worker's `public_addr`.

A multi-hop configuration is when two or more workers are connected, and there are multiple “hops” from a worker to

> the controller.

Reviewer 2: ```suggestion
the controller. There are no limits on the amount of workers allowed in a multi-hop configuration.
```

The multi-hop configuration introduces the concepts of “upstream” and “downstream” workers. If you view controllers
as the “top” of a multi-hop chain, downstream workers reside below a worker in the chain, while upstream workers reside

> above a worker in the chain. For example, int the diagram below, Worker2’s upstream is Worker1, and its

Reviewer: ```suggestion
above a worker in the chain. For example, in the diagram below, Worker2’s upstream is Worker1, and its
```

Reviewer: Sorry I missed this typo in the initial review.

downstream is Worker3.

![multi-hop workers](/img/multihop-workers.png)

You can deploy multi-hop workers in scenarios where inbound network traffic is not allowed. A worker in a private
network can send outbound communication to its upstream worker, and create a reverse proxy to establish a session.

You can configure [target worker filters][] with multi-hop workers to allow for fine-grained control on which workers handle

> ingress and egress for session traffic to a [target][]. Ingress worker filters<sup>HCP Only</sup> determine which

Reviewer 2: ```suggestion
ingress and egress for session traffic to a [target][]. Ingress worker filters determine which
```
Removing the tag as it is moved to the subtitle

```
[kms workers]: /docs/configuration/worker/kms-worker
[target]: /docs/concepts/domain-model/targets
[target worker filters]: /docs/concepts/filtering/worker-tags

> ## Tutorial

Reviewer 2: @Dan-Heath can you include a link to the multihop tutorial when that is available?

Reviewer: Created a task in Asana and assigned it to myself so that I wouldn't forget to do this:

https://app.asana.com/0/563192436488770/1203697401211314/f
